core/src/components/footer/footer.tsx

```diff
@@ -94,6 +113,9 @@ export class Footer implements ComponentInterface {
   render() {
     const { translucent, collapse } = this;
     const mode = getIonMode(this);
+    const tabs = this.el.closest('ion-tabs');
+    const tabBar = tabs?.querySelector('ion-tab-bar');
```

Comment:
Do we need to query for this every render, or are we able to grab it in `connectedCallback`?

Reply:
Querying on every render seems safer since it lets us react to a tab bar being added dynamically later, but I don't feel that strongly about it if you'd rather take the better performance.